docs/investigation/composed-action/REAMDE.md

```diff
@@ -0,0 +1,163 @@
+        input-parameters:
+          raw:
+            data: |
+              name: mattermost
+              owner: mattermost
+    install:
+      depends: ["create-db"]
+      interface:
+        path: "cap.interface.productivity.mattermost.install"
+      input:
+        postgresql:
+          from: psql.postgresql
+        database:
+          from: create-db.database
+        user:
+          from: create-user.user
+  advancedRendering:
+    enabled: false
+  run: false
+  dryRun: false
+  cancel: false
+```
+
+In the future, we can also add a native support for direct Implementation references in steps. When user compose an Action, they already know what they want to achieve. Instead of creating a complex Action Policy, specific Implementations can be selected with an implementation-specific input attached to them. In this way, the Interfaces would be a nice grouping feature used on UI and later user by browsing a catalog of available service, can compose a given Action. Such Action will be easier to rationalize and requires less cognitive overhead as it describes in a declarative way want will be done.
+
+Still, the Global Policy can be used in this approach. When user browse a catalog of available service, only those which conforms with a global policy are displayed.
+
+### Persistence
+
+It could be useful to share a ComposedAction between users. To do so, we can add a new entity under Public Hub tree called `template`:
+
+```
+└── manifests
+   ├── attribute
+   ├── core
+   ├── implementation
+   ├── interface
+   ├── template # holds templates
+   ├── type
+   └── vendor
+```
+
+The new `Template` kind will hold any data indexed by its kind. For example:
+
+```yaml
+ocfVersion: 0.0.1
+revision: 0.1.0
+kind: Template
+metadata:
+  name: config
+  prefix: cap.type.analytics.elasticsearch
+spec:
+  kind: ComposedAction # indexable
+  value:
+    apiVersion: core.capact.io/v1alpha1
+    kind: ComposedAction
+    metadata:
+      name: "full-spec"
+    spec:
+      steps:
+        # ...trimmed...
+```
+
+In the future, this can also be used to share other manifest, where dedicated Implementation and Interface are simply overhead.
+
+## Implementation
+
+The implementation of the Composed Action can be split into two stages.
+
+**MVP scope**
+- Implement CR ComposedAction Operator:
+	- use [Kubebuilder](https://book.kubebuilder.io/) to bootstrap a new project,
+	- update CI,
+	- create a new Helm chart,
+	- ignore the `depends` property and run all steps in sequence.
+	- add minimal E2E tests coverage.
+- Update CLI to support running ComposedAction CR,
+- Document a new feature on website.
+
+**Follow-ups**
+- Add support for parallel and sequence execution,
+- Add support for error handling and retries,
+- Add support to store templates in Public Hub,
+- Add support for direct Implementation references in steps,
+- Add support to create ComposedAction via Dashboard:
+  - Later a dedicated block builder can be added.
+
+### Summary
```

Comment:
I think it will be more convenient if you also copy-paste the Decision from the PR description into this doc 🙂